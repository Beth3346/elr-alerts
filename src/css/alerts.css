.elr-alert {
  padding: 0.625rem 1.25rem;
  max-width: 100%;
  position: relative;
  color: #fff;
  background-color: #333;
  border: 1px solid #333;
}

.elr-alert::after,
.elr-alert::before {
  content: '';
  display: table;
}

.elr-alert::after {
  clear: both;
}

.elr-alert p {
  padding: 0.625rem 0;
}

.elr-alert .close {
  width: 1.25rem;
  height: 1.25rem;
  font-size: 0.875rem;
  padding: 0;
  margin: 0;
  position: absolute;
  right: 0.3125rem;
  top: 0.1875rem;
  border: 0;
  background: 0 0;
  -webkit-transition: color, 0.3s;
  -o-transition: color, 0.3s;
  transition: color, 0.3s;
  color: #fff;
}

.elr-alert .close:hover {
  cursor: pointer;
}

.elr-alert-info {
  background-color: #7dcef6;
  border-color: #1daaf0;
  color: #0c7db5;
}

.elr-alert-danger {
  background-color: #fb9391;
  border-color: #f7322f;
  color: #d10b08;
}

.elr-alert-warning {
  background-color: #e4e087;
  border-color: #d1cb34;
  color: #969222;
}

.elr-alert-success {
  background-color: #bdddbe;
  border-color: #7aba7b;
  color: #4f9850;
}

.elr-alert-muted {
  background-color: #ccc;
  border-color: #999;
  color: #737373;
}
