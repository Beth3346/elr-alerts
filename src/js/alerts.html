<div class="elr-alert-holder">
  <div class="elr-alert elr-success-alert js-dismissible-alert">
    <button class="close js-close-alert">x</button>
    Sucessful form submission!!
  </div>
</div>
