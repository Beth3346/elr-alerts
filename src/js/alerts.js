import $ from 'jquery'
;('use strict')

const createAlert = (type, message) => {
  const className = `elr-alert elr-${type}-alert js-dismissible-alert`
  return $('<div></div>', {
    html: `<p class="elr-alert-message">${message}</p>`,
    class: className
  })
}

const createCloseButton = () => {
  return $('<button></button>', {
    text: 'x',
    class: 'close js-close-alert'
  })
}

export default {
  showAlert(type, message, $holder) {
    const $newAlert = createAlert(type, message)
    const $close = createCloseButton()

    $close.prependTo($newAlert)

    const $alert = $newAlert.prependTo($holder)

    return $alert
  },
  init() {
    $('body').on('click', `.js-close-alert`, function(e) {
      e.preventDefault()
      $(this)
        .parent()
        .fadeOut(300, function() {
          $(this).remove()
        })
    })
  }
}
