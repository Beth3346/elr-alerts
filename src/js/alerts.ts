import * as $ from 'jquery';

const createAlert = (type: string, message: string): JQuery => {
  const className = `elr-alert elr-alert-${type} js-dismissible-alert`;
  return $('<div></div>', {
    html: `<p class="elr-alert-message">${message}</p>`,
    class: className
  });
};

const createCloseButton = (): JQuery => {
  return $('<button></button>', {
    text: 'x',
    class: 'close js-close-alert'
  });
};

export default {
  showAlert(type: string, message: string, $holder: JQuery): JQuery {
    const $newAlert = createAlert(type, message);
    const $close = createCloseButton();

    $close.prependTo($newAlert);

    const $alert = $newAlert.prependTo($holder);

    return $alert;
  },
  init() {
    $('body').on('click', `.js-close-alert`, function(e: Event) {
      e.preventDefault();
      $(this)
        .parent()
        .fadeOut(300, function() {
          $(this).remove();
        });
    });
  }
};
