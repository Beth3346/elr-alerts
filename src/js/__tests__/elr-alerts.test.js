import $ from 'jquery';
import alerts from '../alerts.ts';
import fs from 'fs';
import path from 'path';

const alertHtml = path.join(__dirname, '..', 'alerts.html');
const html = fs.readFileSync(alertHtml).toString();

beforeEach(() => {
  document.body.innerHTML = html;
  alerts.init();
});

describe('init', () => {
  it('should remove alert when close is clicked', done => {
    expect($('.js-dismissible-alert').length).toBe(1);

    $('.js-dismissible-alert')
      .find('.js-close-alert')
      .click();

    setTimeout(() => {
      expect($('.js-dismissible-alert').length).toBe(0);
      done();
    }, 1000);
  });
});
describe('showAlert', () => {
  it('should display an alert', () => {
    const $alert = alerts.showAlert(
      'danger',
      'this is a test',
      $('.elr-alert-holder')
    );

    expect($alert.html()).toMatchSnapshot();
  });
  it('should have a message', () => {
    const message = 'danger! danger!! danger!!!';
    alerts.showAlert('danger', message, $('.elr-alert-holder'));

    expect($('.elr-alert-danger p').text()).toBe(message);
  });
  it('should have a close button', () => {
    alerts.showAlert('danger', 'a message', $('.elr-alert-holder'));
    expect($('.elr-alert-danger').find('button.close').length).toBe(1);
  });
  it('should remove alert when close is clicked', done => {
    alerts.showAlert('danger', 'a message', $('.elr-alert-holder'));

    $('.elr-alert-danger')
      .find('.js-close-alert')
      .click();

    setTimeout(() => {
      expect($('.elr-alert-danger').length).toBe(0);
      done();
    }, 1000);
  });
});
