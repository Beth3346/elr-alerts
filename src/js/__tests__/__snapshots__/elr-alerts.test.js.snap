// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`showAlert should display an alert 1`] = `"<button class=\\"close js-close-alert\\">x</button><p class=\\"elr-alert-message\\">this is a test</p>"`;
